src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -421,7 +448,72 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     }
   }
 
-  override def handleCallRestarts(restartableWorkflows: Seq[RestartableWorkflow])(implicit ec: ExecutionContext): Future[Any] = Future("FIXME")
+  /**
+   * <ul>
+   *   <li>Any execution in Failed should fail the restart.</li>
+   *   <li>Any execution in Aborted should fail the restart.</li>
+   *   <li>Scatters in Starting should fail the restart.</li>
+   *   <li>Collectors in Running should be set back to NotStarted.</li>
+   *   <li>Calls in Starting should be rolled back to NotStarted.</li>
+   *   <li>Calls in Running with no job key should be rolled back to NotStarted.</li>
+   * </ul>
+   *
+   * Calls in Running *with* a job key should be left in Running.  The WorkflowActor is responsible for
+   * resuming the call actors for these calls.
+   */
+  override def prepareForRestart(restartableWorkflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
```

Comment:
I'm not going to raise too much of a ruckus but this feels long/dense to me. Don't restyle just to appease me but could you at least take a pass through to see if there's anything which would make sense to break down?